docs/hub/models-uploading.md

```diff
@@ -6,6 +6,46 @@ You can link repositories with an individual, such as [osanseviero/fashion_brand
 
 There are several ways to upload models to the Hub, described below. We suggest adding a [Model Card](./model-cards) to your repo to document your model.
 
+## Using the Mixin class
```

Comment:
I wonder if this should be the first entry. Not saying it shouldn't, just unsure.

Reply:
It's a good point. Personally it's the cleanest for me, I hope everyone uses this rather than dumping checkpoints in a single repo, but it's up for debate.

Reply:
I think you are right. In the future we could mention the built-in mechanisms in many libraries, but this is better than web or git upload 👍

Reply:
Section content is good but I would not make it the first entry/preferred way of uploading models. I feel that the Hub documentation is meant for any user of the Hub so we should start with the most "universal" upload method and then describe the more specific ones. Since the `PyTorchModelHubMixin` class is restricted to Python+Pytorch users, I would put it under a "Python" section.

Following this spirit, I think it would make more sense to have:

- Uploading models
  - Using the web interface _[same content]_
  - Using the CLI **_[new content]_**
  - Using the Python client library _[same content + link to [upload guide](https://huggingface.co/docs/huggingface_hub/guides/upload) specifically]_
    - Using the Mixin class **_[new content]_**
  - Using the JS client library **_[new content]_**
  - Using Git _[same content]_

(putting git as the last one given it's really not what we aim for)

WDYT?

Reply:
I would not show the most technically challenging option first; that will intimidate people and reduce number of people uploading models IMO. 

Maybe let's add a sentence to the intro paragraph explaining the advantages of this approach + link to it from https://huggingface.co/docs/hub/models-download-stats , but I think showing web interface first makes sense.